### PR TITLE
Add root option to config

### LINKED ISSUE #135: Bring more config options into YAML (such as the Brimcap root)
A community user had been working off the Brim wiki [Remote Workspaces (v0.25.0+)](https://github.com/brimdata/brim/wiki/Remote-Workspaces-%28v0.25.0-%29) article, employing the technique of using Brimcap to locally index the pcaps that corresponded to logs that had been loaded into a remote Workspace, since this makes the **Packets** button work in the app. They then asked this question in a [message in a public Slack thread](https://brimsec.slack.com/archives/C010DR0HHMF/p1630112512015900?thread_ts=1630084715.014300&cid=C010DR0HHMF) (edited slightly for clarity):

> Is it possible to configure the "brimcap-root" path for brimcap when it is being used by the Brim app. If this was possible I presume it wouldn't be necessary for me to index the pcaps twice, since the brimcap-root directory on the zed server side is being shared over NFS and would be accessible to my local brim workstation...

There's a couple ways I can see that we could go about this. Since this represents a configurable setting, one approach that came to mind is Brimcap's Config YAML, since the Brim app already has the ability to point to a custom YAML config via a setting in **Preferences** (see the Brimcap wiki article for [Custom Brimcap Config](https://github.com/brimdata/brimcap/wiki/Custom-Brimcap-Config)). However, at the moment, only `brimcap analyze` uses the config YAML for things like pointing to analyzers, shaper configs, etc. In discussing this with @mattnibs, he agreed that we could expand the YAML to cover the other config options used by other commands such as what's currently only specified via `-root` for  `index` and `search`.

While discussing this, @mattnibs had a couple additional thoughts of what we could cover at the same time, and these sound like fine ideas to me also.

1.  We could have an environment variable that points to your config location
2.  I'd also like a `config` command that prints out the current config. This would be helpful just to see what the default config looks like.

### PR DESCRIPTION
**Pull request by mattnibs**
This is an overhaul of how config works for brimcap:

1. A 'root' option has been added to config that specifies the location of
   the brimcap root. If the -config flag is specified, or the environment
   variable BRIMCAP_CONFIG is set, the brimcap search and index commands
   will look at this value for the location brimcap root.
2. A required 'name' field has been added to an analyzer's config. The
   'name' field is required and must be unique. It is used as a selector
   for command line arguments that can override config values
   (e.g. -analyzer.zeek.stderr).

Closes #135

### CODE REVIEW

File: cli/configflags.go
Comment (nwt) [inline cli/configflags.go:58]:
```suggestion
	for args := os.Args; len(args) > 0; {
```

File: cmd/brimcap/config/command.go
Comment (nwt) [inline cmd/brimcap/config/command.go:16]: Fix this.

**Review by nwt — APPROVED** [review]

**Comment by philrz** [comment]
I was revisiting the [Custom Brimcap Config](https://github.com/brimdata/brimcap/wiki/Custom-Brimcap-Config) wiki article so I could test things out and get ahead of the changes I'd need to make to it once this PR merges. While repeating a command that was already there that used to work ok, I triggered a panic.

```
$ brimcap -version
Version: v1.0.3-10-g20859f4

$ brimcap analyze -config zeek-suricata.yml sample.pcap
brimcap flag redefined: analyzer..cmd
panic: brimcap flag redefined: analyzer..cmd

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc000080d20, {0xdefd10, 0xc0002132c0}, {0xc0002af1a0, 0xd}, {0xc5d0fe, 0xe})
	/usr/lib/go-1.17/src/flag/flag.go:879 +0x2f4
flag.(*FlagSet).StringVar(...)
	/usr/lib/go-1.17/src/flag/flag.go:762
github.com/brimdata/brimcap/analyzer.(*Config).SetFlags(0xc0002132a8, 0xc0002d2048)
	/home/phil/brimcap/analyzer/config.go:35 +0xbd
github.com/brimdata/brimcap/cli.(*ConfigFlags).SetFlags(0xc0002d2018, 0xdefc48)
	/home/phil/brimcap/cli/configflags.go:20 +0x65
github.com/brimdata/brimcap/cmd/brimcap/analyze.New({0xde7960, 0xc00007f5c0}, 0x7)
	/home/phil/brimcap/cmd/brimcap/analyze/command.go:59 +0xcc
github.com/brimdata/zed/pkg/charm.parse(0x12e50e0, {0xc0000320b0, 0x4, 0x4}, {0x0, 0x0})
	/home/phil/go/pkg/mod/github.com/brimdata/zed@v0.29.1-0.20210907143309-882418925884/pkg/charm/instance.go:60 +0x2e5
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0x0, {0xc0000320b0, 0x4, 0x4})
	/home/phil/go/pkg/mod/github.com/brimdata/zed@v0.29.1-0.20210907143309-882418925884/pkg/charm/charm.go:61 +0x36
main.main()
	/home/phil/brimcap/cmd/brimcap/main.go:19 +0x5b
```

**Review by philrz — CHANGES_REQUESTED** [review]
I've tested out the changes in this branch. Putting aside the panic I cited in my prior comment, I did manage to witness the "happy path" of the `BRIMCAP_CONFIG` env variable being obeyed for both `brimcap analyze` and `brimcap index`. in the case of `brimcap index` I also witnessed the `BRIMCAP_ROOT` env variable having the intended effect. Finally, I also set the top-level `root:` config in my YAML and saw `brimcap index` obeying that setting to populate its root.

Regarding that last point, I couldn't seem to find a ztest that exercised the top-level `root:` setting in the YAML, so I only figured it out by looking at the code and guessing. @mattnibs, would you be willing to add a test?

Also, as we discussed at the team sync this morning, Brim app changes are going to be required since those `name:` settings will be required in the YAML that it creates on the fly. @mattnibs are you going to make those changes yourself or coordinate with @mason-fish and/or @jameskerr such that they do it? Since we manually advance Brim's pointer to Brimcap I guess there's no risk of CI breaking the moment this merges on the Brimcap side. But it'd probably be good to get this lined up sooner than later.

**Comment by mattnibs** [comment]
@philrz I believe I have addressed your feedback.

**Comment by mattnibs** [comment]
I'll start working on the brim side changes, I won't merge this until that is ready.

**Review by philrz — APPROVED** [review]

File: cmd/brimcap/migrate/command.go
Comment (philrz) [inline cmd/brimcap/migrate/command.go:303]: @mattnibs: Just wanted to draw your attention to this change...

**Pull request merged**
